### PR TITLE
[NPU] Enable automatic ascend_attn selection for vision attention and graph runners

### PR DESCRIPTION
**Pull request by ltaodream**
## Background

SGLang already provides and registers `VisionAscendAttention` as the optimized multimodal attention implementation for Ascend NPU. However, unlike CUDA, ROCm, MUSA, XPU, and CPU, Ascend NPU did not have a platform-specific default in `VisionAttention._determine_attention_backend()`.

As a result, NPU users had to explicitly pass:

```bash
--mm-attention-backend ascend_attn
```

Without this argument, multimodal models using the common `VisionAttention` path fell back to `sdpa` and eventually executed `torch.nn.functional.scaled_dot_product_attention`, even though an optimized Ascend implementation was available.

## Problem

There were two related issues:

1. `VisionAttention` did not select `ascend_attn` by default on NPU, so omitting the CLI argument caused an unexpected SDPA fallback.
2. Simply adding an NPU default was not sufficient for ViT graph execution. The CUDA, InternVL CUDA, and NPU ViT graph runners read `ServerArgs.mm_attention_backend` again instead of using the backend already resolved by `VisionAttention`. When the user omitted the CLI argument, the raw server argument remained `None`, so graph capture could reject the backend even though the attention module had selected a valid platform default.

The expected behavior is that backend selection follows the existing priority:

```text
explicit server argument > model/constructor argument > platform default
```

and that eager execution and graph runners use the same resolved backend.

## Changes

This PR:

- Selects `ascend_attn` as the platform default for `VisionAttention` on Ascend NPU.
- Stores the effective backend name on each `VisionAttention` instance after applying the server override, constructor argument, and platform default.
- Updates the generic CUDA ViT graph runner, InternVL CUDA graph runner, and NPU ViT graph runner to consume the resolved backend from the attention module instead of re-reading the raw CLI argument.
- Improves unsupported-backend errors in the graph runners by including the effective backend name.
- Adds unit coverage for NPU backend-selection priority and graph-runner backend propagation.

This does not mutate the global `ServerArgs.mm_attention_backend`. Explicit user and model overrides therefore keep their current precedence, and model-specific attention selectors outside the common `VisionAttention` path remain unchanged.

## User impact

For multimodal models using the common `VisionAttention` implementation, Ascend users no longer need to specify `--mm-attention-backend ascend_attn`. Omitting the option now consistently selects the optimized Ascend backend in both eager and supported ViT NPU graph execution.

### Ascend hardware test

Tested on a real Ascend 910B2C NPU with a local `Qwen3VLForConditionalGeneration` checkpoint. The server was launched without `--mm-attention-backend`.

The server arguments confirmed that no override was provided:

```text
mm_attention_backend=None
```

The model then selected the Ascend backend automatically:

```text
Multimodal attention backend not set. Use ascend_attn.
Using ascend_attn as multimodal attention backend.
```

Results:

- Eager multimodal image generation completed successfully with HTTP 200.
- Multimodal prefill executed successfully on NPU without entering the SDPA fallback.
- ViT NPU graph was enabled with `SGLANG_VIT_ENABLE_CUDA_GRAPH=1`, still without specifying `--mm-attention-backend`.
- The first image request successfully created/captured the ViT graph and completed in approximately 6.14 seconds.
- A second request with the same shape successfully reused/replayed the graph and completed in approximately 0.17 seconds.
- Both requests returned HTTP 200 with consistent output and reported `#cached-token: 0`.

If the NPU graph runner had continued to read the raw `None` server argument, graph creation would have failed with an unsupported-backend error. Successful capture and replay verify that the resolved `ascend_attn` backend is propagated to the graph runner.

### Unit tests

```bash
PYTHONPATH=python pytest -q \
  test/registered/unit/layers/attention/test_vision_backend_selection.py \
  test/registered/unit/layers/attention/test_vision_seqlens.py \
  test/registered/unit/layers/attention/test_vision_max_seqlen.py
```

```text
13 passed
```

Additional checks:

- Python compilation checks passed for all modified and newly added Python files.
- `git diff --check` passed.

---
### CI States

Latest PR Test (Base): :x: [Run #30696705688](https://github.com/sgl-project/sglang/actions/runs/30696705688)
Latest PR Test (Extra): :white_check_mark: [Run #30699490862](https://github.com/sgl-project/sglang/actions/runs/30699490862)

### CODE REVIEW

File: python/sglang/srt/layers/attention/vision.py [hunk 1]
```diff
@@ -1167,6 +1174,8 @@ def _determine_attention_backend(self, passed_backend: Optional[str]) -> str:
                 backend = "fa4"
             else:
                 backend = "triton_attn"
+        elif _is_npu:
+            backend = "ascend_attn"
```
Comment (JustinTong0323) [inline python/sglang/srt/layers/attention/vision.py:1190]: The unconditional NPU default drops masks for every masked `VisionAttention` caller: `VisionAscendAttention.forward()` ignores `attention_mask`, while Mllama passes its aspect-ratio/padding mask and CLIP text passes a causal mask that SDPA previously honored. Keep these callers on SDPA or implement mask support in the Ascend backend, and add a masked-attention regression test before changing the platform default.

**Review by JustinTong0323 — COMMENTED** [review]
The NPU default currently breaks masked `VisionAttention` callers.

File: python/sglang/srt/layers/attention/vision.py (hunk 1 above)
Comment (ltaodream) [reply python/sglang/srt/layers/attention/vision.py:1190]: Thanks for catching this. Fixed in ed19574dc.VisionAscendAttention now falls back to SDPA whenever an explicit attention_mask is provided, preserving Mllama’s aspect-ratio/padding mask and CLIP’s causal mask. Unmasked inputs continue to use the fused Ascend backend.We also added regression tests for causal and padding masks, validating the outputs against SDPA and ensuring the fused operator is not called for masked inputs. A separate test confirms that unmasked inputs still use the fused Ascend path.The masked and unmasked paths were additionally validated with FP16 and BF16 on an Ascend 910B2C.

**Review by JustinTong0323 — APPROVED** [review]
LGTM.
